.github/CONTRIBUTING.md

```diff
@@ -210,6 +210,84 @@ cmake -S pybind11/ -B build
 cmake --build build
 ```
 
+### Explanation of the SDist/wheel building design
+
+> These details are _only_ for building from the Python sources from git. The
```

Comment:
This intro statement is already a bit confusing -- "building Python sources" seems like an oxymoron. :). I know what's meant, but this whole section could be a bit beginner-friendly (more on that also in the next comments). You have to keep in mind that many pybind11 users come from the C++/CMake world and aren't Python packaging experts.